akka-http-core/src/main/scala/akka/http/engine/ws/Masking.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.engine.ws
+
+import akka.stream.scaladsl.Flow
+import akka.stream.stage.{ SyncDirective, Context, StageState, StatefulStage }
+
+import scala.util.Random
+
+object Masking {
+  def maskIf(condition: Boolean): Flow[FramePart, FramePart, Unit] =
+    if (condition) Flow[FramePart].transform(() ⇒ new Masking())
+    else Flow[FramePart]
+  def unmaskIf(condition: Boolean): Flow[FramePart, FramePart, Unit] =
+    if (condition) Flow[FramePart].transform(() ⇒ new Unmasking())
+    else Flow[FramePart]
+
+  class Masking extends Masker {
+    val random = new Random()
```

Comment:
Yes, this is not yet finished. This is security-relevant (needs sufficient entropy by spec) and needs to be configurable and have a sane default.